### ISSUE TITLE
v1.1 e2e fails against v1.3 cluster (master & nodes): secret file has wrong mode

### ISSUE THREAD

**Issue by ihmccreery** [kubernetes/kubernetes#26662]
```
12:49:20 • Failure [9.107 seconds]
12:49:20 Secrets
12:49:20 /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/secrets.go:100
12:49:20   should be consumable from pods [Conformance] [It]
12:49:20   /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/secrets.go:99
12:49:20 
12:49:20   "mode of file \"/etc/secret-volume/data-1\": -r--r--r--" in container output
12:49:20   Expected
12:49:20       <string>: mode of file "/etc/secret-volume/data-1": Lrwxrwxrwx
12:49:20       content of file "/etc/secret-volume/data-1": value-1
12:49:20       
12:49:20       
12:49:20   to contain substring
12:49:20       <string>: mode of file "/etc/secret-volume/data-1": -r--r--r--
12:49:20 
12:49:20   /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/util.go:1251
```

View logs and artifacts at https://console.cloud.google.com/storage/browser/kubernetes-jenkins/logs/kubernetes-e2e-gke-1.1-1.3-upgrade-cluster/209

**Comment by ihmccreery** [comment]
@pmorie Can you take a look at this?  Any idea why this would be happening?

You can repro by turning up a 1.3 cluster and running 1.1 tests against it.  Let me know if you have trouble.

**Comment by thockin** [comment]
@jingxu97 for your queue.  Note that the behavior of secrets has changed, and that the new file *is* a link.  Ike, how are we supposed to handle teh case where the 1.1 e2e is asserting something that is no longer true for 1.3 ?

**Comment by ihmccreery** [comment]
I can handle this, if it's intended behavior.  We can either version-guard the logic somehow (there's no precedent for version-guarding on node version—not sure yet if/how we would do that), or, better, make the test properly resilient to this behavior.  Should we really be checking ACLs at all?

**Comment by pmorie** [comment]
@ihmccreery seems like we could check the status of the node the pod is scheduled to and retrieve the version from that, but that seems ugly.

**Comment by pmorie** [comment]
As @thockin said, the behavior has changed.  Can I clarify, this is the 1.1 e2e binary running against a 1.3 cluster?  Want to make sure we did not miss any changes we should have made.

**Comment by jingxu97** [comment]
Hi,

Could anyone help me point out some information about setting up the
cluster 1.3 version and run 1.1 tests? Thanks a lot!

Best
Jing

On Sat, Jun 4, 2016 at 11:11 AM, Paul Morie <notifications@github.com>
wrote:

> As @thockin <https://github.com/thockin> said, the behavior has changed.
> Can I clarify, this is the 1.1 e2e binary running against a 1.3 cluster?
> Want to make sure we did not miss any changes we should have made.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/issues/26662#issuecomment-223769849>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ASSNxV5CnWdIRJKAS23DE5DYL1Fu9uR5ks5qIb_IgaJpZM4IsBKF>
> .
>



-- 
- Jing

**Comment by thockin** [comment]
What's next for this?  Is this on Ike to figure out how to guard the assertion across versions?  I don't think there's actually a prod-code bug here.

Ike, FTR, where are docs on running cross-version tests manually?

**Comment by jingxu97** [comment]
I set up the environment and run the test, this failed test mentioned in this issue is working for for my test. 
should be consumable from pods [Conformance] [It]
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/secrets.go:100

But there are some other failures. I dont think they are related.

```
• Failure [14.567 seconds]
[k8s.io] ConfigMap
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/framework.go:639
  should be consumable from pods in volume as non-root with FSGroup [Feature:FSGroup] [It]
  /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/configmap.go:45
  Expected error:
      <*errors.errorString | 0xc821ef6bf0>: {
          s: "pod 'pod-configmaps-e0cefe7c-2e91-11e6-a241-5065f351b2ea' terminated with failure: &{ExitCode:-1 Signal:0 Reason:ContainerCannotRun Message: StartedAt:{Time:2016-06-09 15:31:16 -0700 PDT} FinishedAt:{Time:2016-06-09 15:31:16 -0700 PDT} ContainerID:docker://271b63551e74433e5f7bd5c749daea603707394bb55a7aac5438beff8c605fae}",
      }
      pod 'pod-configmaps-e0cefe7c-2e91-11e6-a241-5065f351b2ea' terminated with failure: &{ExitCode:-1 Signal:0 Reason:ContainerCannotRun Message: StartedAt:{Time:2016-06-09 15:31:16 -0700 PDT} FinishedAt:{Time:2016-06-09 15:31:16 -0700 PDT} ContainerID:docker://271b63551e74433e5f7bd5c749daea603707394bb55a7aac5438beff8c605fae}
  not to have occurred
 /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/util.go:1941
```

**Comment by saad-ali** [comment]
So since it sounds like like expected behavior, I say the test is probably testing the wrong fields if we are breaking the behavior across versions. But just my 2c.

Removing @jingxu97 from assignees since it doesn't look like there's any further debugging needed here.

**Comment by jingxu97** [comment]
For running cross-version tests, I have some steps listed in issue #27096 https://github.com/kubernetes/kubernetes/issues/27096

**Comment by ihmccreery** [comment]
Sorry about the delay.

> What's next for this? Is this on Ike to figure out how to guard the assertion across versions? I don't think there's actually a prod-code bug here.

I agree with @saad-ali; why are we checking ACLs at all; can we just remove that completely?

> Ike, FTR, where are docs on running cross-version tests manually?

Currently, there are none.  I filed #27224 and will try to get to it asap.

**Comment by erictune** [comment]
@ihmccreery does this need a new owner?

**Comment by ihmccreery** [comment]
It ought to have a new owner, yes.  Assigning to you to find a new owner.

**Comment by jingxu97** [comment]
fixed by #29859
